debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/gccheck/CheckEngine.java

```diff
@@ -1195,28 +1195,32 @@ private int checkJ9Object(J9ObjectPointer object, GCHeapRegionDescriptor regionD
 			}
 		}
 
-		if (J9BuildFlags.env_data64 && isIndexableDataAddressFlagSet() && ObjectModel.isIndexable(object)) {
-
-			boolean isCorrectData = true;
-			J9IndexableObjectPointer array = J9IndexableObjectPointer.cast(object);
-			try {
-				if (J9IndexableObjectHelper.isInlineContiguousArraylet(array)) {
-					isCorrectData = J9IndexableObjectHelper.getDataAddrForContiguous(array).equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.contiguousHeaderSize())));
-				} else {
-					isCorrectData = J9IndexableObjectHelper.getDataAddrForDiscontiguous(array).equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.discontiguousHeaderSize())));
+		try {
+			if (J9BuildFlags.env_data64 && isIndexableDataAddressFlagSet() && ObjectModel.isIndexable(object)) {
+				if (!_javaVM.isIndexableDataAddrPresent().isZero()) {
+					J9IndexableObjectPointer array = J9IndexableObjectPointer.cast(object);
+					UDATA dataSizeInBytes = ObjectModel.getDataSizeInBytes(array);
+					VoidPointer dataAddr = J9IndexableObjectHelper.getDataAddrForIndexable(array);
+					boolean isCorrectDataAddrPointer;
+					if (dataSizeInBytes.isZero()) {
+						isCorrectDataAddrPointer = (dataAddr.isNull() || dataAddr.equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.discontiguousHeaderSize()))));
+					} else if (dataSizeInBytes.lt(_javaVM.arrayletLeafSize())) {
+						isCorrectDataAddrPointer = dataAddr.equals(VoidPointer.cast(array.addOffset(J9IndexableObjectHelper.contiguousHeaderSize())));
```

Comment:
Lines 1206 and 1208 deserve to be folded.

Reply:
If by folded, it means the same as separated into separate lines, then I have done this in the changes that will be pushed shortly.